src/app/components/home/components/ip-summary/ip-summary.component.ts

```diff
@@ -1,11 +1,108 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnChanges } from '@angular/core';
+import { AbuseIpService } from '@/app/services/abuse-ip.service';
+import { NgIf, NgStyle } from '@angular/common';
+import { trigger, transition, style, animate } from '@angular/animations';
+import { NotFoundIconComponent } from '@/app/components/icons/not-found-icon/not-found-icon.component';
+import { ErrorIconComponent } from '@/app/components/icons/error-icon/error-icon.component';
+
+export interface ErrorObj {
+  val: boolean;
+  msg?: string;
+  code?: number;
+}
+
+export interface SummaryObj {
+  countryName: string;
+  countryCode: string;
+  isp: string;
+  domain: string;
+  abuseConfidenceScore: number;
+  totalReports: number;
+  lastReportedAt: string;
+}
+
+export interface SummaryRes {
+  data: SummaryObj;
+}
 
 @Component({
   selector: 'app-ip-summary',
   standalone: true,
-  imports: [],
+  imports: [NgIf, NgStyle, NotFoundIconComponent, ErrorIconComponent],
   templateUrl: './ip-summary.component.html',
+  animations: [
+    trigger('fadeInOut', [
+      transition(':enter', [style({ opacity: 0 }), animate('800ms 800ms', style({ opacity: 1 }))]),
+      transition(':leave', [animate(800, style({ opacity: 0 }))]),
+    ]),
+  ],
 })
-export class IPSummaryComponent {
+export class IPSummaryComponent implements OnChanges {
   @Input() ip = '';
+  summary!: SummaryObj;
+  isLoaded!: boolean;
+  error: ErrorObj = {
+    val: false,
+  };
+
+  constructor(private service: AbuseIpService) {}
+
+  ngOnChanges(): void {
+    this.isLoaded = false;
+
+    this.error = {
+      val: false,
+    };
+
+    const response = this.service.check(this.ip);
+
+    response.subscribe({
+      next: (data) => {
+        const dt = data as SummaryRes;
+        this.summary = dt.data as SummaryObj;
+
+        this.isLoaded = true;
+
+        console.log(this.summary);
+      },
+      error: (err) => {
+        console.log(err);
+
+        this.summary = {} as SummaryObj;
+
+        this.error = {
+          val: true,
+          msg: err.message,
+          code: err.status,
+        };
+
+        this.isLoaded = true;
+
+        console.log(this.error);
+      },
+    });
+  }
+
+  getDate(strDate: string) {
+    if (!strDate || strDate.length === 0) {
+      return '';
+    } else {
+      const d = new Date(strDate);
+      return d.toLocaleDateString('en-GB');
```

Comment:
Deveria ser pt-BR aqui?
```suggestion
      return d.toLocaleDateString('pt-BR');
```